File: src/main/scala/laoflch/debezium/connector/informix/InformixTopicSelector.scala
package laoflch.debezium.connector.informix

import io.debezium.relational.TableId
import io.debezium.schema.TopicSelector


/**
 * The topic naming strategy based on connector configuration and table name
 *
 * @author laoflch Luo
 *
 */
object InformixTopicSelector {
  def defaultSelector(connectorConfig: InformixConnectorConfig): TopicSelector[TableId] =
    TopicSelector.defaultSelector(connectorConfig,
      (tableId: TableId, prefix: String, delimiter: String) =>
        String.join(delimiter, prefix, tableId.schema, tableId.table))
}

File: src/main/java/laoflch/debezium/connector/informix/InformixTopicSelector.java
package laoflch.debezium.connector.informix;

import io.debezium.relational.TableId;
import io.debezium.schema.TopicSelector;

public class InformixTopicSelector {
    public static TopicSelector<TableId> defaultSelector(InformixConnectorConfig connectorConfig) {
        return TopicSelector.defaultSelector(connectorConfig,
                (tableId, prefix, delimiter) -> String.join(delimiter, prefix, tableId.schema(), tableId.table()));
    }
}
